aspnetcore/migration/1x-to-2x.md

```diff
@@ -0,0 +1,198 @@
+author: scottaddie
+description: Migrating an ASP.NET Core 1.x project to ASP.NET Core 2.x
+keywords: ASP.NET Core, migrating
+ms.author: scaddie
+manager: wpickett
+ms.date: 07/19/2017
+ms.topic: article
+ms.assetid: 8468d859-ff32-4a92-9e62-08c4a9e36594
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: migration/1x-to-2x
+---
+# Migrating From ASP.NET Core 1.x to 2.x
+
+By [Scott Addie](https://github.com/scottaddie)
+
+This article shows how to migrate an existing ASP.NET Core 1.x project to ASP.NET Core 2.x. Non-breaking changes are outside of the article's scope.
+
+## Prerequisites
+Install the following prerequisites before migrating to ASP.NET Core 2.x:
+- Visual Studio 2017 Preview version 15.3 or later, Visual Studio Code, or Visual Studio for Mac
+- .NET Core 2.x or .NET Framework 4.6.1+
+
+For applications hosted on Windows Server with IIS and Kestrel, the [.NET Core Windows Server Hosting bundle](xref:publishing/iis) must be updated.
+
+> [!NOTE]
+> Thanks to .NET Standard 2.0, .NET Core 2.0 offers a much larger surface area than .NET Core 1.x. If you're targeting .NET Framework solely because of missing APIs in .NET Core 1.x, targeting .NET Core 2.0 is likely to work for you now.
+
+## Target Framework Moniker (TFM)
+Projects targeting .NET Core must use the TFM of a version greater than or equal to .NET Core 2.0:
+
+[!code-xml[Main](../migration/1x-to-2x/samples/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App.csproj?range=3)]
+
+Projects targeting .NET Framework must use the TFM of a version greater than or equal to .NET Framework 4.6.1:
+
+```xml
+<TargetFramework>net461</TargetFramework>
+```
+
+## global.json
+If the solution relies upon a [*global.json*](https://docs.microsoft.com/dotnet/core/tools/global-json) file to target a specific .NET Core SDK version, update it to use the desired version installed on the machine:
+
+[!code-json[Main](../migration/1x-to-2x/samples/AspNetCoreDotNetCore2.0App/global.json?highlight=3)]
+
+## PackageReference
+In a 1.x *.csproj* file, an assortment of NuGet packages is needed:
+
+[!code-xml[Main](../migration/1x-to-2x/samples/AspNetCoreDotNetCore1.1App/AspNetCoreDotNetCore1.1App/AspNetCoreDotNetCore1.1App.csproj?range=9-26)]
+
+In an ASP.NET Core 2.x project targeting .NET Core 2.x, a single [meta-package](xref:fundamentals/metapackage) reference in the *.csproj* file replaces the aforementioned collection of packages:
+
+[!code-xml[Main](../migration/1x-to-2x/samples/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App/AspNetCoreDotNetCore2.0App.csproj?range=9-11)]
+
+All the features of ASP.NET Core 2.x and Entity Framework Core 2.x are included in the meta-package.
+
+In an ASP.NET Core 2.x project targeting .NET Framework, the meta-package is incompatible. An upgrade of each NuGet package reference to 2.x is required.
+
+## DotNetCliToolReference
+Update the `Version` attributes of each `<DotNetCliToolReference />` node to the desired 2.x version:
```

Comment:
I'm not crazy about desired. Could we just day 
node to the ~desired~ 2.x version